fix: change `Component.Evidence.Identity` to array

DmitriyLewen opened the pull request:
## Description
in CycloneDX 1.6 type of `Component.Evidence.Identity` has been changed to `[]Identity`.
`Identity` as `struct` was marked as `deprecated`.
https://cyclonedx.org/docs/1.6/json/#tab-pane_metadata_component_evidence_identity_oneOf_i0

## Related issues
- Close #192

DmitriyLewen commented on convert.go:
```diff
@@ -195,6 +195,13 @@ func convertEvidence(c *Component, specVersion SpecVersion) {
 	}
 
 	if specVersion < SpecVersion1_6 {
+		// Spec version 1.5 uses only one Identity.
+		// cf. https://cyclonedx.org/docs/1.5/json/#components_items_evidence_identity
+		if c.Evidence.Identity != nil {
+			ids := *c.Evidence.Identity
+			ids = ids[:1]
+			c.Evidence.Identity = &ids
+		}
```
I'm not sure about this way...
Maybe we need to select one `Identifier` using `Field` with some order

Please tell me if you have a better solution.

nscuro replied:
IMO that is fine for now, since it keeps the insertion order.

DmitriyLewen commented on cyclonedx_xml_test.go:
I overwrite all the Evidence fields in the `MarshalXML` and `UnmarshalXML` functions, so I added test cases for each field.

Tell me if this is too much

nscuro commented:
Hmmm, seems like CI never ran for his PR and I can't manually trigger it. @DmitriyLewen do you mind re-opening this PR, or alternatively amend your last commit so GitHub triggers a build?

DmitriyLewen commented:
Hi @nscuro
I reran actions (I merged `master` branch), but this action still doesn't run despite having your approval - https://github.com/CycloneDX/cyclonedx-go/actions/runs/12270387292
Do you have ideas?

nscuro commented:
@DmitriyLewen I had to approve CI to run since this is your first PR for this repo. However, I wasn't given that option before you pushed a new commit, which is odd.

Now we have a few test failures.

DmitriyLewen commented:
Hello @nscuro 
Thanks!

I fixed [TestEvidence_MarshalJSON/WithIdentify](https://github.com/CycloneDX/cyclonedx-go/actions/runs/12270387292/job/34251289295#step:4:84) in https://github.com/CycloneDX/cyclonedx-go/pull/204/commits/24e9503293f0837e6e7ea3ff670ef958e6075b87

About [TestRoundTripJSON/valid-evidence.json](https://github.com/CycloneDX/cyclonedx-go/actions/runs/12270387292/job/34251289295#step:4:363):
I forgot to update `valid-evidence.json` and updated this file in https://github.com/CycloneDX/cyclonedx-go/pull/204/commits/d68a199bc1747e5d6a7d4196c2f270535bbf6e3e.
But this test is still fail (because schema for `identity` has been changed). 
What can i do with this test?

DmitriyLewen commented:
Got it. Thanks for your help!

All tests passed:
```
PASS
coverage: 83.0% of statements
ok      github.com/CycloneDX/cyclonedx-go       37.215s

```

nscuro reviewed (approved):
Thanks!
